File: Medium/Java/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Initialize 2 variables to keep track of the sum
        int sum = nums[0];
        int tempSum = sum;

        // Use a for loop to see what the largest subarray is
        for(int i = 1; i < nums.length; i++){
            tempSum = Math.max(tempSum + nums[i], nums[i]);
            sum = Math.max(tempSum, sum);
        }
        
        // Return the largest subarray sum
        return sum;
    }
}

File: Medium/Python/53. Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        # Initialize 2 variables to keep track of the sum
        maxSum = nums[0]
        tempSum = maxSum

        # Use a for loop to see what the largest subarray is
        for i in range(1, len(nums)):
            tempSum = max(tempSum + nums[i], nums[i])
            maxSum = max(tempSum, maxSum)
        
        # Return the largest subarray sum
        return maxSum
